### PR TITLE
remove LocalEnv unused fields & LocalAppDetector

### PR DESCRIPTION
**Pull request by sqs**
Just code cleanup and unused code removal.

## Test plan

CI

### CODE REVIEW

**Comment by sqs** [comment]
@sourcegraph/jetbrains-plugin Will this cause problems with the JetBrains plugin? It removes unused fields in the webview protocol.

**Review by abeatrix — APPROVED** [review]
thank you!

File: agent/src/vscode-shim.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ import type { ClientInfo, ExtensionConfiguration } from './protocol-alias'
 // Not using CODY_TESTING because it changes the URL endpoint we send requests
 // to and we want to send requests to sourcegraph.com because we record the HTTP
 // traffic.
-const isTesting = process.env.CODY_SHIM_TESTING === 'true'
+const isTesting = typeof process !== 'undefined' && process.env.CODY_SHIM_TESTING === 'true'
```
Comment (dominiccooney) [inline agent/src/vscode-shim.ts:58]: Why do we need this guard?

If this is to put Agent in web workers or VSCode Web maintenance, I'm reticent to add another dimension of complexity to this right now. For example, to fix Enterprise proxy issues with non-VSCode, we will need to lean on the node networking stack... symf and cody-engine need subprocess... etc.

**Review by dominiccooney — COMMENTED** [review]
IDEs do depend on the webview messages in general, but I believe JetBrains will be fine with this change. One question inline.

File: agent/src/vscode-shim.ts (hunk 1 above)
Comment (sqs) [reply agent/src/vscode-shim.ts:58]: Yeah, it is to try putting agent in a web worker and VS Code Web maintenance. That's fine, I understand, I will take the `typeof process !== 'undefined'` guards out of this PR.

Comment (dominiccooney) [reply agent/src/vscode-shim.ts]: Thank you. I am keen to support VSCode Web, and experiment with Agent in workers, once we have JetBrains on a sure footing.

**Review by dominiccooney — APPROVED** [review]
Looks good! Thank you for the clean up.

**Pull request merged**